src/main/java/net/smartcosmos/dao/things/util/UuidUtil.java

```diff
@@ -19,7 +19,7 @@
 
     public static UUID getUuidFromUrn(String urn) throws IllegalArgumentException {
 
-        String urnScheme = "urn:.*:uuid:([A-F0-9]{8}-[A-F0-9]{4}-[A-F0-9]{4}-[A-F0-9]{4}-[A-F0-9]{12})";
+        String urnScheme = "^urn:.*:uuid:([A-F0-9]{8}-[A-F0-9]{4}-[A-F0-9]{4}-[A-F0-9]{4}-[A-F0-9]{12})$";
```

Comment:
Well, you could consider it part of it:

Previously, URNs like `urn:thing:uuid:926a94f9-ab5c-448c-9109-c3d7dff2cf87NotValid` or  `This-is-not-a-valid-URN:urn:thing:uuid:926a94f9-ab5c-448c-9109-c3d7dff2cf87` did not throw any exception, because such values were considered valid URNs, but they actually are not.